receiver/snowflakereceiver/factory.go

```diff
@@ -13,3 +13,64 @@
 // limitations under the License.
 
 package snowflakereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/snowflakereceiver"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/snowflakereceiver/internal/metadata"
+)
+
+const (
+	typeStr         = "snowflakemetrics"
+	defaultInterval = 30 * time.Minute
+	defaultRole     = "ACCOUNTADMIN"
+	defaultDB       = "SNOWFLAKE"
+	defaultSchema   = "ACCOUNT_USAGE"
+)
+
+func createDefaultConfig() component.Config {
+	return &Config{
+		ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+			CollectionInterval: defaultInterval,
+		},
+		Metrics:  metadata.DefaultMetricsSettings(),
+		Schema:   defaultSchema,
+		Database: defaultDB,
+		Role:     defaultRole,
+	}
+}
+
+func NewFactory() component.Factory {
+	return receiver.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		receiver.WithMetrics(createMetricsReceiver, component.StabilityLevelAlpha),
```

Comment:
Let's keep the stability level as Development as it's documented in the README. We can bump it as a separate follow-up PR

Reply:
Sorry I did not see development as an enum on my editor! That's what I get for being lazy.